app/src/main/java/ch/epfl/sweng/erpa/services/UserAuthService.java

```diff
@@ -0,0 +1,43 @@
+package ch.epfl.sweng.erpa.services;
+
+import com.annimon.stream.Optional;
+
+import java.nio.charset.StandardCharsets;
+
+import javax.inject.Inject;
+
+import at.favre.lib.crypto.bcrypt.BCrypt;
+import ch.epfl.sweng.erpa.model.UserAuth;
+import ch.epfl.sweng.erpa.services.dummy.DummyRemoteServicesProvider;
+
+public class UserAuthService {
+
+    private RemoteServicesProvider rsp;
+
+    @Inject public UserAuthService() {
+        rsp = new DummyRemoteServicesProvider();
+    }
+
+    public Optional<UserAuth> getUserAuth(String username, String password) {
+        Optional<String> uid = rsp.getUidFromUsername(username);
+        if (uid.isPresent()) {
+            String id = uid.get();
+            String accessToken = createAccessToken(id, password);
+            if(rsp.verifyAccessToken(id, accessToken)) {
+                return Optional.of(new UserAuth(id, accessToken));
+            }
+        }
+        return Optional.empty();
+    }
+
+    private String createAccessToken(String uid, String password) {
+        byte[] uidBytes = uid.getBytes(StandardCharsets.UTF_8);
```

Comment:
Since UID it's 128 bit-long[^1] and you need 16 bytes (128 bit), could you please properly convert the UID into a salt?

[^1]: https://en.wikipedia.org/wiki/Universally_unique_identifier